prereise/gather/solardata/helpers.py

```diff
@@ -32,23 +40,31 @@ def to_reise(data):
     return profile
 
 
-def get_plant_info_unique_location(plant):
-    """Identify unique location and return relevant information of plants at
-    location.
+def get_plant_id_unique_location(plant):
+    """Identify unique location among plants.
 
     :param pandas.DataFrame plant: plant data frame.
-    :return: (*dict*) -- keys are coordinates of location. Values is a list of
-        2-tuple giving the plant id at location along with its capacity.
+    :return: (*dict*) -- keys are coordinates. Values is a list of *'plant_id'*.
+    :raises TypeError: if *'plant'* is not a data frame.
+    :raises ValueError: if *'plant_id'* is not the index and/or *'lat'* and *'lon'* are
+        not among the columns.
     """
+    if not isinstance(plant, pd.DataFrame):
+        raise TypeError("plant must be a pandas.DataFrame")
+    if not (plant.index.name == "plant_id" and {"lat", "lon"}.issubset(plant.columns)):
+        raise ValueError(
+            "data frame must have plant_id as index and lat and lon among columns"
+        )
+
     n_target = len(plant)
 
     # Identify unique location
     coord = OrderedDict()
     for i in range(n_target):
         key = (str(plant.lon.values[i]), str(plant.lat.values[i]))
         if key not in coord.keys():
-            coord[key] = [(plant.index[i], plant.Pmax.values[i])]
+            coord[key] = [plant.index[i]]
         else:
-            coord[key].append((plant.index[i], plant.Pmax.values[i]))
+            coord[key].append(plant.index[i])
```

Comment:
Why do we return the coordinates as tuples of strings?

Why don't we get the mapping of coordinates to plants using a simple pandas method? E.g. `coord = plant.groupby(["lon", "lat"]).groups`

Reply:
I did that a long time ago when I did not know much about `pandas`. I can refactor it.

Reply:
Done in 14e979a96958e1ffd516475d6ab95007df531c3b